src/screens/WebviewScreen/WebviewScreen.tsx

```diff
@@ -1,35 +1,60 @@
-import React from 'react';
-import WebView from 'react-native-webview';
+import React, { useRef, useState } from 'react';
+import WebView, { WebViewNavigation } from 'react-native-webview';
+import { ProgressBar } from 'react-native-paper';
 
-import { Appbar } from '@components';
 import { useTheme } from '@hooks/persisted';
 import { WebviewScreenProps } from '@navigators/types';
 import { getUserAgent } from '@hooks/persisted/useUserAgent';
-import { isUrlAbsolute } from '@plugins/helpers/isAbsoluteUrl';
-import { getPlugin } from '@plugins/pluginManager';
+import { resolveUrl } from '@services/plugin/fetch';
+import Appbar from './components/Appbar';
 
 const WebviewScreen = ({ route, navigation }: WebviewScreenProps) => {
+  const { name, url, pluginId, isNovel } = route.params;
+  const uri = pluginId ? resolveUrl(pluginId, url, isNovel) : url;
```

Comment:
because this is potentially error, I think try catch is needed.
```ts
const [uri, setUri] = useState(url);

useEffect(() => {
 try{
  // do smt
 } catch(e) {
  // do smt
 }
},[]);
```

Reply:
It might be better to return a try catch construct in the resolveUrl function

Reply:
yea